source/directives/rhField.js

```diff
@@ -25,24 +25,118 @@ module.exports = [ "$compile", function ( $compile )
                 return scope.def.type;
             };
 
+            var getPattern = function ()
+            {
+                if( scope.def.pattern )
+                {
+                    return scope.def.pattern;
+                }
+                if( getType() === "email" )
+                {
+                    return "^.+@.+\\..{2,}$";
+                }
+                if( getType() === "url" )
+                {
+                    return "^.+://.+\\..{2,}$";
+                }
+                if( getType() === "tel" )
+                {
+                    return ".*\\d.*\\d.*\\d.*\\d.*\\d.*\\d.*\\d.*\\d.*";
```

Comment:
I found that there are practically speaking no international phone numbers less than 8 digits so this is just a loose match for that. I'm open to making it stricter, but not too strict. If a project wanted something stricter they could also just add a `pattern` to their field definition.